code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -430,8 +430,9 @@
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
 /datum/reagent/medicine/calomel/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
-	for(var/datum/reagent/toxin/R in M.reagents.reagent_list)
-		M.reagents.remove_reagent(R.type, 3 * REM * delta_time)
+	for(var/datum/reagent/R in M.reagents.reagent_list)
+		if(istype(R.type, /datum/reagent/medicine/calomel))
```

Comment:
```suggestion
		if(!istype(R.type, /datum/reagent/medicine/calomel))
```

Reply:
Currently this just purges itself.

Reply:
im convinced vscode is intentionally inverting my code

Reply:
Also, you are typechecking the iterator reagent's type, not the iterator reagent itself.

Reply:
This is what happens to a mf when he changes if(R.type != /datum/reagent/medicine/calomel) at the last second